### ISSUE TITLE
🐛 [Bug Report] mind controlled target should take initial aggro

### ISSUE THREAD

**Issue by confessore** [cmangos/issues#2968]
### Bug Details

mind controlled target should take initial aggro

i am a player and i am running in a group dungeon and i try to mind control a target but all of the mobs run towards me instead of fighting the mind controlled target

### Steps to Reproduce

1. be a priest
2. go to ramparts with your friends
3. mind control a target in a group
4. have all the mobs ignore the mind controlled target and rush towards you

### Expected behavior

mind controlled target should take initial aggro

### Suggested Workaround

not yet

### Core SHA1 Commit Hash

HEAD

### Database SHA1 Commit Hash

HEAD

### Operating System

ubuntu 20.04

### Client Version

2.4.3 (The Burning Crusade)

**Comment by DrKittens** [comment]
>Expected behavior
>>mind controlled target should take initial aggro

I dont believe this is how it worked in TBC as mind control still generates hate which is shared by the pack, do you have a video demonstrating it not aggro-ing grouped enemies in Vanilla/TBC OR mind control falling off a target and that target not attacking the player / been aggro'd after the fact?

IE when mind control ends on a single target that target will still attack you, the fact they were charmed for the duration doesn't change that. The target you've mind controlled has no aggro / threat until it does anything at which point the pack priority should change.

The only time i can recall it working as described is when mind controlling a mob that is not grouped but simply in range of another mob with an empty threat table

**Comment by killerwife** [comment]
I also do not believe this is how it should work. Please provide proof

**Comment by confessore** [comment]
> I dont believe this is how it worked in TBC as mind control still generates hate which is shared by the pack, do you have a video demonstrating it not aggro-ing grouped enemies in Vanilla/TBC OR mind control falling off a target and that target not attacking the player / been aggro'd after the fact?
> 
> 
> 
> IE when mind control ends on a single target that target will still attack you, the fact they were charmed for the duration doesn't change that. The target you've mind controlled has no aggro / threat until it does anything at which point the pack priority should change.
> 
> 
> 
> The only time i can recall it working as described is when mind controlling a mob that is not grouped but simply in range of another mob with an empty threat table

allow us time to create a video of various behaviors

i believe my understanding to be flawed



initial aggro should go to the priest casting mind control

initial aggro should go to the mind controlled mob from mob_pack0 that pulls mob_pack1

and i'll edit this comment when we have produced sample of actual behavior

**Comment by AnonXS** [comment]
Still needs evidence, though i think you're on to something, problem is mob being hit by negative spell (the mc hitting the mob) will trigger the init aggro of the surrounding npcs into the priest instead of the initaggro onto the mind controlled mob.
